ui/pages/confirm-transaction-base/confirm-transaction-base.component.js

```diff
@@ -408,114 +413,130 @@ export default class ConfirmTransactionBase extends Component {
-              subText={t('editGasSubTextFee', [
-                <b key="editGasSubTextFeeLabel">
-                  {t('editGasSubTextFeeLabel')}
-                </b>,
-                <div
-                  key="editGasSubTextFeeValue"
-                  className="confirm-page-container-content__currency-container"
-                >
-                  {renderHeartBeatIfNotInTest()}
-                  <UserPreferencedCurrencyDisplay
-                    key="editGasSubTextFeeAmount"
-                    type={PRIMARY}
-                    value={hexMaximumTransactionFee}
-                    hideLabel={!useNativeCurrencyAsPrimaryCurrency}
-                  />
-                </div>,
-              ])}
-              subTitle={
-                <>
-                  {txData.dappSuggestedGasFees ? (
-                    <Typography
-                      variant={TYPOGRAPHY.H7}
-                      fontStyle={FONT_STYLE.ITALIC}
-                      color={COLORS.UI4}
-                    >
-                      {t('transactionDetailDappGasMoreInfo')}
-                    </Typography>
+            EIP_1559_V2 ? (
+              <GasDetailsItem
+                key="gas_details"
+                hexMaximumTransactionFee={hexMaximumTransactionFee}
+                hexMinimumTransactionFee={hexMinimumTransactionFee}
+                isMainnet={isMainnet}
+                maxFeePerGas={maxFeePerGas}
+                maxPriorityFeePerGas={maxPriorityFeePerGas}
+                supportsEIP1559={supportsEIP1559}
+                txData={txData}
+                useNativeCurrencyAsPrimaryCurrency={
+                  useNativeCurrencyAsPrimaryCurrency
+                }
+              />
+            ) : (
+              <TransactionDetailItem
+                key="gas-item"
+                detailTitle={
+                  txData.dappSuggestedGasFees ? (
+                    <>
+                      {t('transactionDetailGasHeading')}
+                      <InfoTooltip
+                        contentText={t('transactionDetailDappGasTooltip')}
+                        position="top"
+                      >
+                        <i className="fa fa-info-circle" />
+                      </InfoTooltip>
+                    </>
                   ) : (
-                    ''
-                  )}
-                  {supportsEIP1559 && (
-                    <GasTiming
-                      maxPriorityFeePerGas={hexWEIToDecGWEI(
-                        maxPriorityFeePerGas ||
-                          txData.txParams.maxPriorityFeePerGas,
-                      )}
-                      maxFeePerGas={hexWEIToDecGWEI(
-                        maxFeePerGas || txData.txParams.maxFeePerGas,
-                      )}
+                    <>
+                      {t('transactionDetailGasHeading')}
+                      <InfoTooltip
+                        contentText={
+                          <>
+                            <p>
```

Comment:
Can we use our `<Typography>` component here instead of native `p` tags?

Reply:
@darkwing : this is existing code that will go away once flag `EIP-1559-V2` is removed.